File: Data Structures & Algorithms/non-overlapping-intervals/submission-2.py
class Solution:
    def eraseOverlapIntervals(self, intervals: List[List[int]]) -> int:
        if len(intervals) <= 1: return 0
        intervals.sort(key=lambda x: x[0])
        prevEnd = intervals[0][1]
        cnt = 0
        for s, e in intervals[1:]:
            if s >= prevEnd:
                prevEnd = e
            else:
                cnt += 1
                prevEnd = min(prevEnd, e)
        return cnt

File: Data Structures & Algorithms/palindromic-substrings/submission-2.cpp
class Solution {
public:
    int countSubstrings(string s) {
        int n = s.size();
        vector<vector<bool>> map(n, vector<bool>(n, false));
        int cnt = 0;

        for(int i = n-1; i >= 0; i--) {
            for(int j = i; j < n; j++) {
                if((s[i] == s[j]) && ((j-i < 3) || (map[i+1][j-1]))) {
                    map[i][j] = true;
                    cnt++;
                }
            }
        }
        return cnt;
    }
};
